packages/infinite-scroll/src/index.ts

```diff
@@ -0,0 +1,141 @@
+    cb: InfiniteScrollCallback
+    onScroll: () => void
+    observer?: MutationObserver
+  }
+}
+
+const getScrollOptions = (el: HTMLElement, instance: ComponentPublicInstance): ScrollOptions => {
+  return entries(attributes)
+    .reduce((acm, [name, option]) => {
+      const { type, default: defaultValue } = option
+      const attrVal = el.getAttribute(`infinite-scroll-${name}`)
+      let value = instance[attrVal] ?? attrVal ?? defaultValue
+      value = value === 'false' ? false : value
+      value = type(value)
+      acm[name] = isNaN(value) ? defaultValue : value
+      return acm
+    }, {} as ScrollOptions)
+}
+
+const destroyObserver = (el: InfiniteScrollEl) => {
+  const { observer } = el[SCOPE]
+
+  if (observer) {
+    observer.disconnect()
+    delete el[SCOPE].observer
+  }
+}
+
+const handleScroll = (el: InfiniteScrollEl, cb: InfiniteScrollCallback) => {
+  const { container, containerEl, instance, observer } = el[SCOPE]
+  const { disabled, distance } = getScrollOptions(el, instance)
+
+  // trigger only if full check has done and not disabled
+  if (observer || disabled) return
+
+  const { clientHeight, scrollHeight, scrollTop } = containerEl
+  let shouldTrigger = false
+
+  if (container === el) {
+    shouldTrigger = scrollHeight - (clientHeight + scrollTop) <= distance
+  } else {
+    // get the scrollHeight since el might be visible overflow
+    const { clientTop, scrollHeight: height } = el
+    const offsetTop = getOffsetTopDistance(el, containerEl)
+    shouldTrigger = scrollTop + clientHeight >= offsetTop + clientTop + height - distance
+  }
+
+  if (shouldTrigger) {
+    cb.call(instance)
+  }
+}
+
+function checkFull(el: InfiniteScrollEl, cb: InfiniteScrollCallback) {
+  const { containerEl, instance } = el[SCOPE]
+  const { disabled } = getScrollOptions(el, instance)
+
+  if (disabled) return
+
+  if (containerEl.scrollHeight <= containerEl.clientHeight) {
+    cb.call(instance)
+  } else {
+    destroyObserver(el)
+  }
+}
+
+const InfiniteScroll: ObjectDirective<InfiniteScrollEl, InfiniteScrollCallback> = {
+  async mounted(el, binding) {
+    const { instance, value: cb } = binding
+
+    if (!isFunction(cb)) {
+      throwError(SCOPE, '\'v-infinite-scroll\' binding value must be a function')
+    }
+
+    // ensure parentNode mounted
+    await nextTick()
+
+    const { delay, immediate } = getScrollOptions(el, instance)
+    const container = getScrollContainer(el, true)
+    const containerEl = container === window ? document.documentElement : (container as HTMLElement)
+    const onScroll = throttle(handleScroll.bind(null, el, cb), delay)
+
+    if (!container) return
+
+    el[SCOPE] = { instance, container, containerEl, cb, onScroll }
+
+    if (immediate) {
+      const observer = new MutationObserver(throttle(checkFull.bind(null, el, cb), CHECK_INTERVAL))
+      el[SCOPE].observer = observer
+      observer.observe(el, { childList: true, subtree: true })
+      checkFull(el, cb)
+    }
```

Comment:
What happens if `immediate` is `false`?

Reply:
The purpose of `infinite-scroll-immediate` is whether to execute the load method immediate in order to avoid that the scroll container is not overflow. If `immediate` is `false`, no such check will be done. The overflow state of scroll container all depends on outsiders.

Reply:
I have checked the rest of your code, the only place you use `observer.observe` is here, if this is just a single call, I think I'm ok with this